python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -1932,11 +2028,30 @@ def _remeasure_star_background(self, result, background_to_photometric_ratio=Non
             background=result.background,
             backgroundToPhotometricRatio=background_to_photometric_ratio,
         ).background
-        # Isolate the final pedestal background to log the computed value
+        # Isolate the final pedestal background to log the computed value. Set
+        # minimum and maximum allowable to avoid unnaturally extreme values
+        # (this step is largely trying to correct for oversubtraction, so the
+        # minimum value set is much looser than the maximum).
         pedestalBackground = afwMath.BackgroundList()
         pedestalBackground.append(pedestalBackgroundList[1])
         pedestalBackgroundLevel = pedestalBackground.getImage().array[0, 0]
-        self.log.warning("Subtracted pedestalBackgroundLevel = %.4f", pedestalBackgroundLevel)
+
+        maxPedestal = self.config.star_background_max_pedestal*abs(median_background)
+        if pedestalBackgroundLevel > maxPedestal:
+            self.log.warning("Computed final background pedestal (%.4f) was larger than the "
+                             "maximum allowed (%.2f).  Setting it to %.2f",
+                             pedestalBackgroundLevel, maxPedestal, maxPedestal)
+            pedestalBackgroundLevel = self.config.star_background_max_pedestal
+
+        minPedestal = -1.0*self.config.star_background_min_pedestal*abs(median_background)
+        self.log.debug("minPedestal = %.3f  maxPedestal = %.3f", minPedestal, maxPedestal)
+        if pedestalBackgroundLevel < minPedestal:
+            self.log.warning("Computed final background pedestal (%.4f) was smaller than the "
+                             "minimum allowed (%.2f).  Setting it to %.2f",
+                             pedestalBackgroundLevel, minPedestal, minPedestal)
+            pedestalBackgroundLevel = minPedestal
+
+        self.log.info("Subtracted pedestalBackgroundLevel = %.4f", pedestalBackgroundLevel)
```

Comment:
So ... I'm confused how this even works now.  Does this change the value inside the background list? 
```
        pedestalBackground = afwMath.BackgroundList()
        pedestalBackground.append(pedestalBackgroundList[1])
        pedestalBackgroundLevel = pedestalBackground.getImage().array[0, 0]
```
And then ... it's already been subtracted by the task ... so this won't work.